openconnect_sso/app.py

```diff
@@ -28,17 +29,50 @@ def run(args):
     try:
         if os.name == "nt":
             asyncio.set_event_loop(asyncio.ProactorEventLoop())
-        return asyncio.get_event_loop().run_until_complete(_run(args))
+        auth_response, selected_profile = asyncio.get_event_loop().run_until_complete(
+            _run(args)
+        )
     except KeyboardInterrupt:
         logger.warn("CTRL-C pressed, exiting")
+        return 1
```

Comment:
Previously CTRL-C was not regarded as an error, it exited gracefully. I can see the value of returning an error code for it. AFAIK it should be 130 (`-2` in twos complement, or 256+signal.SIGINT casted to a byte) for CTRL-C

Reply:
My thinking here was that if you abort with CTRL-C during the initial part of the script (before we launch openconnect), then a non-zero exit code is warranted, because one cannot argue that the program has completed successfully in any way, when the user aborted authentication and we never got around to even starting `openconnect`.

Conversely, once `openconnect` is running, pressing CTRL-C is a natural way to disconnect the VPN session (is there an easier way for the user to signal the intent to disconnect?), so CTRL-C then should not by regarded as an error. You'll see this towards the end of `run()` where I handle `KeyboardInterrupt` with `return 0`.

That said, I will change the exit code in the first case from `1` to `130`.